src/zenml/zen_server/routers/projects_endpoints.py

```diff
@@ -225,69 +218,69 @@ def list_project_stacks(
         project_name_or_id: Name or ID of the project.
         user_name_or_id: Optionally filter by name or ID of the user.
         component_id: Optionally filter by component that is part of the stack.
-        stack_name: Optionally filter by stack name
+        name: Optionally filter by stack name
         is_shared: Optionally filter by shared status of the stack
-        hydrated: Defines if stack components, users and projects will be
-                  included by reference (FALSE) or as model (TRUE)
+        auth_context: Authentication Context
 
     Returns:
         All stacks part of the specified project.
     """
-    stacks_list = zen_store().list_stacks(
+    stacks = zen_store().list_stacks(
         project_name_or_id=project_name_or_id,
-        user_name_or_id=user_name_or_id,
+        user_name_or_id=user_name_or_id or auth_context.user.id,
         component_id=component_id,
-        is_shared=is_shared,
-        name=stack_name,
+        is_shared=False,
+        name=name,
     )
-    if hydrated:
-        return [stack.to_hydrated_model() for stack in stacks_list]
-    else:
-        return stacks_list
+    # In case the user didn't explicitly filter for is shared == False
+    if is_shared is None or is_shared:
+        shared_stacks = zen_store().list_stacks(
+            project_name_or_id=project_name_or_id,
+            user_name_or_id=user_name_or_id or auth_context.user.id,
+            component_id=component_id,
+            is_shared=True,
+            name=name,
+        )
+        stacks += shared_stacks
```

Comment:
Result of discussion with @bcdurak: it makes sense in the server since we only need it when running through a server. In the local case, there is only one user, so this logic is not needed.